test/test_transforms_v2_refactored.py

```diff
@@ -1634,3 +1637,143 @@ def test_transform_negative_degrees_error(self):
+        batch_size = 32
+        H, W = 12, 12
+
+        preproc = transforms.Compose([transforms.PILToTensor(), transforms.ToDtype(torch.float32)])
+        if one_hot:
+
+            class ToOneHot(torch.nn.Module):
+                def forward(self, inpt):
+                    img, label = inpt
+                    return img, torch.nn.functional.one_hot(label, num_classes=num_categories)
+
+            preproc = transforms.Compose([preproc, ToOneHot()])
+
+        dataset = FakeData(size=batch_size, image_size=(3, H, W), num_classes=num_categories, transforms=preproc)
+        if T == "Compose":
+            cutmix = transforms.Cutmix(alpha=0.5, num_categories=num_categories)
+            mixup = transforms.Mixup(alpha=0.5, num_categories=num_categories)
+            cutmix_mixup = transforms.Compose([cutmix, mixup])
+            expected_num_non_zero_labels = 3  # see common_checks
+        else:
+            cutmix_mixup = T(alpha=0.5, num_categories=num_categories)
+            expected_num_non_zero_labels = 2  # see common_checks
+
+        dl = DataLoader(dataset, batch_size=batch_size)
+
+        # Input sanity checks
+        img, target = next(iter(dl))
+        assert isinstance(img, torch.Tensor) and isinstance(target, torch.Tensor)
+        assert target.shape == (batch_size, num_categories) if one_hot else (batch_size,)
+
+        def check_output(img, target):
+            assert img.shape == (batch_size, 3, H, W)
+            assert target.shape == (batch_size, num_categories)
+            torch.testing.assert_close(target.sum(axis=-1), torch.ones(batch_size))
+            # Below we check the number of non-zero values in the target tensor.
+            # When just CutMix() (or just MixUp()) is called, we should expect 2
+            # non-zero label values per sample. Although, it may happen that
+            # only 1 non-zero value is present, basically if the transform had
+            # no effect. Here we make sure that:
+            # - there is at least one sample with 2 non-zero values
+            # - there is no sample with more than 2 non-zero values
+            # When CutMix() and MixUp() are called in sequence together, we
+            # should expect 3 instead of 2. That's the
+            # expected_num_non_zero_labels threshold.
+            num_non_zero_values = (target != 0).sum(axis=-1)
+            assert (num_non_zero_values == expected_num_non_zero_labels).any()
+            assert (num_non_zero_values <= expected_num_non_zero_labels).all()
+            assert (num_non_zero_values > 0).all()  # Note: we already know that from target.sum(axis=-1) check above
+
+        # After Dataloader, as unpacked input
+        img, target = next(iter(dl))
+        assert target.shape == (batch_size, num_categories) if one_hot else (batch_size,)
+        img, target = cutmix_mixup(img, target)
+        check_output(img, target)
+
+        # After Dataloader, as packed input
+        packed_from_dl = next(iter(dl))
```

Comment:
Do we need to have this is "separate" test or can we just do

```python
img, target = packed_from_dl = next(iter(dl))
```

Reply:
Whether both cases should be supported, we can discuss. But right now we're testing both (as they're different). I can add the line you suggested but it's saving one line. Is it necessary?